components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/util/EndpointUtil.java

```diff
@@ -1899,4 +1902,22 @@ private static Class<? extends OAuthAuthzRequest> getOAuthAuthzRequestClass() th
         }
         return oAuthAuthzRequestClass;
     }
+
+    /**
+     * Validate the response mode against the response type as per FAPI spec.
+     * Response mode required to be jwt if code response type is used.
+     * <a href="https://openid.net/specs/openid-financial-api-part-2-1_0.html#authorization-server">5.2.2-2.2</a>
+     *
+     * @param responseType response mode
+     * @param responseMode response type
+     * @throws OAuthProblemException when response mode is not valid
+     */
+    public static void validateFAPIAllowedResponseMode(String responseType, String responseMode)
```

Comment:
Lets rename the method to better fit the validations done. Maybe something like `validateFAPIAllowedResponseTypeAndMode`
```
shall require; 
1. the response_type value code id_token, or
2. the response_type value code in conjunction with the response_mode value jwt;
```

Reply:
Additionally, there should be a unit test to make sure the spec allowed values are covered.

Reply:
fixed with [a2e92b8](https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2157/commits/a2e92b846147834a7532ac6bfe3df2bb9bc6c8b2)